server/src/main/java/org/opensearch/indices/replication/SegmentReplicationState.java

```diff
@@ -76,9 +76,10 @@ public static Stage fromId(byte id) {
 
     private Stage stage;
     private ReplicationLuceneIndex index;
-
     private final ReplicationTimer overallTimer;
-    private final Map<String, Long> timingData;
+
+    // Timing data will have as many entries as stages, plus one
+    private final Map<String, Long> timingData = new ConcurrentHashMap<>(Stage.values().length + 1);
```

Comment:
I think we still want to init this in constructors only and send this map over the wire, otherwise the map will empty on the other end when querying stats.  You can make the map concurrent in the ctor or copy it to write.

Reply:
I am going with copying the map and sending copy of map in output stream.